zkchannels-crypto/src/proofs/range.rs

```diff
@@ -102,6 +69,16 @@ impl RangeConstraintParameters {
     pub fn public_key(&self) -> &PublicKey<1> {
         &self.public_key
     }
+
+    /// Verify the signatures inside the RangeConstraintParameters are valid signatures on the appropriate scalars
+    pub fn validate(&self) -> Result<(), String> {
+        for (i, sig) in self.digit_signatures.iter().enumerate() {
+            if !sig.verify(&self.public_key, &Scalar::from(i as u64).into()) {
+                return Err("The signatures in the RangeConstraintParameters must be valid signatures on the appropriate scalars".to_string());
```

Comment:
Thought: It looks like we use string errors throughout the library, so I would not recommend changing it, but at some point we may find it useful to make named error types. We don't match on the errors anywhere that we use this library, but it seems like a professional, Rusty thing to do.

Reply:
Although on that note, I did, on Mukund's input, create named errors in revlock.rs and found it useful for testing.